Add missing asserts

E.g. to show where argument should be normalized.  See the GMP codebase for inspiration.

Add CI job with NDEBUG unset: `pip install -e . -Csetup-args=-Dbuildtype=debug` // https://mesonbuild.com/meson-python/how-to-guides/debug-builds.html

Hypothesis/pytest bug?

Unhelpful traceback with assert(0):
```
============================= test session starts ==============================
platform linux -- Python 3.13.5, pytest-8.4.1, pluggy-1.6.0 -- /opt/hostedtoolcache/Python/3.13.5/x64/bin/python
  The GNU GMP version: 6.3.0
  Bits per limb:       64
  Size of a limb:      8
cachedir: .pytest_cache
hypothesis profile 'default' -> deadline=timedelta(milliseconds=300000)
rootdir: /home/runner/work/python-gmp/python-gmp
configfile: pyproject.toml
plugins: hypothesis-6.136.0
collecting ... collected 69 items
Fatal Python error: Aborted
Current thread 0x00007f2e3908fb80 (most recent call first):
  File "/home/runner/work/python-gmp/python-gmp/tests/test_functions.py", line 23 in test_isqrt
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/hypothesis/core.py", line 1005 in test
  File "/home/runner/work/python-gmp/python-gmp/tests/test_functions.py", line 22 in test_isqrt
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/hypothesis/core.py", line 1091 in run
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/hypothesis/core.py", line 818 in default_executor
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/hypothesis/core.py", line 1135 in execute_once
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/hypothesis/core.py", line 1195 in _execute_once_for_engine
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/hypothesis/internal/conjecture/engine.py", line 399 in __stoppable_test_function
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/hypothesis/internal/conjecture/engine.py", line 506 in test_function
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/hypothesis/internal/conjecture/engine.py", line 494 in cached_test_function
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/hypothesis/internal/conjecture/engine.py", line 1028 in generate_new_examples
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/hypothesis/internal/conjecture/engine.py", line 1418 in _run
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/hypothesis/internal/conjecture/engine.py", line 854 in run
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/hypothesis/core.py", line 1373 in run_engine
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/hypothesis/core.py", line 2020 in wrapped_test
  File "/home/runner/work/python-gmp/python-gmp/tests/test_functions.py", line 22 in test_isqrt
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/python.py", line 157 in pytest_pyfunc_call
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/python.py", line 1671 in runtest
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/runner.py", line 178 in pytest_runtest_call
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/runner.py", line 246 in <lambda>
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/runner.py", line 344 in from_call
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/runner.py", line 245 in call_and_report
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/runner.py", line 136 in runtestprotocol
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/runner.py", line 117 in pytest_runtest_protocol
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/main.py", line 367 in pytest_runtestloop
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/main.py", line 343 in _main
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/main.py", line 289 in wrap_session
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/main.py", line 336 in pytest_cmdline_main
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/config/__init__.py", line 175 in main
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/config/__init__.py", line 201 in console_main
  File "/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pytest/__main__.py", line 9 in <module>
  File "<frozen runpy>", line 88 in _run_code
  File "<frozen runpy>", line 198 in _run_module_as_main
Extension modules: gmp (total: 1)
/home/runner/work/_temp/345323c5-6f89-4f53-b820-b2cc9632bdb8.sh: line 4: 29453 Aborted                 (core dumped) python -m pytest --hypothesis-profile=default
tests/test_functions.py::test_isqrt 
Error: Process completed with exit code 134.
```
See https://github.com/diofant/python-gmp/actions/runs/16385629817/job/46304690075?pr=204

--- a/.github/workflows/coverage.yml
+++ b/.github/workflows/coverage.yml
@@ -32,7 +32,7 @@ jobs:
           PKG_CONFIG_PATH: .local/lib/pkgconfig
       - name: Build and run coverage tests
         run: |
-          meson setup build --prefix=/usr -Db_coverage=true
+          meson setup build -Db_coverage=true -Ddebug=true
           meson install -C build --destdir ../build-install
           export PYTHONPATH="$(find build-install -name site-packages)"
           python -m pytest --hypothesis-profile=default

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -29,7 +29,9 @@ jobs:
           allow-prereleases: true
       - run: bash scripts/cibw_before_all.sh
       - run: pip install --upgrade pip
-      - run: pip --verbose install --editable .[tests]
+      - run: |
+          pip --verbose install --editable .[tests] \
+              -Csetup-args=-Dbuildtype=debug
         env:
           PKG_CONFIG_PATH: .local/lib/pkgconfig
           CFLAGS: -Wall -Wpedantic -Werror -std=c17
@@ -54,7 +56,8 @@ jobs:
       - run: |
           export CFLAGS="$CFLAGS -I$(brew --prefix gmp)/include"
           export LDFLAGS="-L$(brew --prefix gmp)/lib"
-          pip --verbose install --editable .[tests]
+          pip --verbose install --editable .[tests] \
+              -Csetup-args=-Dbuildtype=debug
       - run: pytest
   windows:
     needs:
@@ -83,5 +86,6 @@ jobs:
           python -m venv .test-venv
           source .test-venv/bin/activate
           python -m pip install --upgrade pip
-          pip --verbose install -e .[tests]
+          pip --verbose install --editable .[tests] \
+              -Csetup-args=-Dbuildtype=debug
           pytest

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -52,7 +52,7 @@ docs = ["sphinx>=8.2"]
 develop = ["python-gmp[tests,docs]", "pre-commit", "pyperf"]
 
 [tool.pytest.ini_options]
-addopts = "--durations=7"
+addopts = "--durations=7 --capture=no"
 xfail_strict = true
 
 [tool.ruff]

